### PR TITLE
MDEXP-199 Updates to the factored version of SRS

### PR DESCRIPTION
**Pull request by Igor-Gorchakov**
https://issues.folio.org/browse/MDEXP-199
## Purpose

- Refactor to use updated API (prepare the PR to be merged the same day as SRS refactoring-master is merged to master)
- Verify the functionality is working on the environment
- Notify Folijet team that SRS can be released

## Approach
1. The SRS leaves using CQL in favour of SQL, so the API is changed
2. Interface source-storage-records 1.0 is updated to source-storage-source-records 2.0, permission is updated too
3. API is updated, no need to retrieve a collection of records using GET request, now SRS provides POST API to retrieve records sending UUIDs in the request body
4. Updated unit tests

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/resources/mockData/srs/get_records_response.json [hunk 1]
```diff
@@ -1,15 +1,9 @@
 {
-  "records": [
+  "sourceRecords": [
     {
-      "id": "bc5caa84-e85c-4c98-9956-50ccdaad7115",
+      "recordId": "bc5caa84-e85c-4c98-9956-50ccdaad7115",
```
Comment (KaterynaSenchenko) [inline src/test/resources/mockData/srs/get_records_response.json:4]: It is irrelevant for mock data, but just so you know, SRS now have constraints that ensure recordId, rawRecordId and parsedRecordId are the same

File: src/test/java/org/folio/rest/MockServer.java [hunk 2]
```diff
@@ -183,17 +183,12 @@ private void handleGetInstanceRecord(RoutingContext ctx) {
...
-      .getParam("query");
     try {
-      JsonObject srsRecords;
-      srsRecords = new JsonObject(RestVerticleTestBase.getMockData(SRS_RECORDS_MOCK_DATA_PATH));
+      JsonObject srsRecords = new JsonObject(RestVerticleTestBase.getMockData(SRS_RECORDS_MOCK_DATA_PATH));
       //fetch the ids from the query and remove them from the mock if not in the request
-      List<String> instanceIds = Arrays.stream(query.split("or"))
-        .map(st -> st.split("==")[1].trim().replace(")", ""))
-        .collect(Collectors.toList());
+      List<String> instanceIds = ctx.getBodyAsJsonArray().getList();
```
Comment (KVupp) [inline src/test/java/org/folio/rest/MockServer.java:189]: I think the part of instanceIds filtering is still needed, That is how integration tests are run based on the request.

Comment (Igor-Gorchakov) [reply src/test/java/org/folio/rest/MockServer.java:189]: The filtering is still in place, I just take instanceIds from request body, not from request url

File: src/test/resources/mockData/srs/get_records_response.json (hunk 1 above)
Comment (Igor-Gorchakov) [reply src/test/resources/mockData/srs/get_records_response.json:4]: Done

**Pull request merged**